azurerm/resource_arm_private_endpoint_test.go

```diff
@@ -0,0 +1,271 @@
+					resource.TestCheckResourceAttrSet(resourceName, "private_link_service_connection.0.private_link_service_id"),
+					resource.TestCheckResourceAttr(resourceName, "private_link_service_connection.0.request_message", "Please approve my connection"),
+				),
+			},
+			{
+				Config: testAccAzureRMPrivateEndpoint_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateEndpointExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "private_link_service_connection.0.group_ids.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "private_link_service_connection.0.request_message", "plz approve my request"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.env", "test"),
+				),
+			},
+			{
+				Config: testAccAzureRMPrivateEndpoint_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateEndpointExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "private_link_service_connection.0.request_message", "Please approve my connection"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMPrivateEndpointExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Private Endpoint not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).network.PrivateEndpointClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		if resp, err := client.Get(ctx, resourceGroup, name, ""); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Private Endpoint %q (Resource Group %q) does not exist", name, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on PrivateEndpointClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMPrivateEndpointDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).network.PrivateEndpointClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_private_endpoint" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name, ""); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on PrivateEndpointClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMPrivateEndpointTemplate_standardResources(rInt int, location string) string {
```

Comment:
COuld we be consistent with the rest of the provider:
```suggestion
func testAccAzureRMPrivateEndpointTemplate_template(rInt int, location string) string {
```

Reply:
Fixed.